### ISSUE TITLE
Extremely high gas fee shown while attempting to deposit DAI in Ethereum

### ISSUE THREAD

**Issue by Patrick1904** [aurora-is-near/rainbow-bridge-frontend#230]
**Problem**
The gas fee shown in Metamask seems unreasonably high. Here is the TX from Explorer: https://explorer.near.org/transactions/GKSFuDXg825h2vXrnLJCntd5arMSUyqczAJziYB169Ku

<img width="1162" alt="Screen Shot 2021-04-28 at 12 09 44 PM" src="https://user-images.githubusercontent.com/24921205/116459422-a1142480-a81a-11eb-8f8e-62d9e11a0261.png">

**Comment by mfornet** [comment]
I think something went wrong estimating the amount of gas required to run your transaction (either that or an unreasonable gas price).

From the following recent txs I'm seeing gas consumed is ~310K, and gas limit (estimated) is ~460K.
- https://etherscan.io/tx/0xf56a70bb04e11f8b111de18d71cb15ae5ad59262fff03d05c6537ae677ccbde2
- https://etherscan.io/tx/0x4aa6512cc48c474b95da43c2399285788aa7982227fdf9059bf686a61768b9d4
- https://etherscan.io/tx/0xa7ac714c6a9b2863e2b1d0f1a70845dcff3993f5b47e6c768296cab3f1eb3ff8

@paouvrard can we modify gas limit to always be minimum between estimated and some constant (let say 500K).
Which tool are we using right now to estimate the gas limit? is it from ethers library? Maybe it is worth to make some research about why can this happen, and whether it is safe to use lower limit.

@Patrick1904 can you still reproduce this now? In that case, can you check what is the gas price and gas limit used. I don't see how to do it from the metamask you shared in screenshot, but the version I'm using has some advanced options that allows me to check gas limit and gas price.

**Comment by paouvrard** [comment]
@mfornet The gas price is estimated by Metamask. I've never experienced the mis estimation, but since this seems to have happened several times we can fix it with a safe margin, like 500K.

**Comment by Patrick1904** [comment]
> @Patrick1904 can you still reproduce this now? In that case, can you check what is the gas price and gas limit used. I don't see how to do it from the metamask you shared in screenshot, but the version I'm using has some advanced options that allows me to check gas limit and gas price.

@mfornet yep I'm still able to reproduce the high gas fee, and here is the gas price and limit used (I haven't changed it):

<img width="1001" alt="Screen Shot 2021-05-05 at 1 43 52 PM" src="https://user-images.githubusercontent.com/24921205/117206802-23b35b80-ada8-11eb-8fc4-078e961bd93d.png">
